src/dataflow-types/src/types.rs

```diff
@@ -1307,42 +1313,115 @@ pub mod sources {
     /// AWS SDK so that we can implement `Serialize` and `Deserialize`.
     #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
     pub struct AwsConfig {
-        /// The optional static credentials to use.
+        /// AWS Credentials, or where to find them
         ///
-        /// If unset, credentials should instead be obtained from the environment.
-        pub credentials: Option<AwsCredentials>,
+        /// * Default uses the standard AWS credentials provider chain
+        /// * Static uses credentials provided via WITH options
+        /// * Profile uses the credentials provided in the specific profile
+        pub credentials: AwsCredentials,
+        /// The AWS role to assume
+        pub role_arn: Option<String>,
+        /// The external_id for this customer for role assumption
+        ///
+        /// <https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html>
+        pub external_id: Option<String>,
         /// The AWS region to use.
-        pub region: String,
+        ///
+        /// Uses the default region (looking at env vars, config files, etc) if not provideed.
+        pub region: Option<String>,
         /// The custom AWS endpoint to use, if any.
         pub endpoint: Option<SerdeUri>,
     }
 
-    /// Static AWS credentials for a source or sink.
+    /// AWS credentials for a source or sink.
     #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
-    pub struct AwsCredentials {
-        pub access_key_id: String,
-        pub secret_access_key: String,
-        pub session_token: Option<String>,
+    pub enum AwsCredentials {
+        /// Look for credentials using the [default provider][docs] chain
+        ///
+        /// [docs]: https://docs.rs/aws-config/latest/aws_config/default_provider/index.html
+        Default,
+        /// Load credentials using the given named profile
+        Profile { profile_name: String },
+        /// Use the credentials provided as WITH options
```

Comment:
```suggestion
        /// Use the enclosed static credentials.
```

(Talking about `WITH` options at this layer doesn't seem right.)